core/src/main/java/io/opencensus/tags/TagContextBinarySerializer.java

```diff
@@ -29,25 +27,24 @@
   /**
    * Serializes the {@code TagContext} into the on-the-wire representation.
    *
-   * <p>This method should be the inverse of {@link #deserialize}.
+   * <p>This method should be the inverse of {@link #fromByteArray}.
    *
    * @param tags the {@code TagContext} to serialize.
-   * @param output the {@link OutputStream} to write the serialized tags to.
-   * @throws IOException if there is an {@code IOException} while writing to {@code output}.
+   * @return the on-the-wire representation of a {@code TagContext}.
    */
-  public abstract void serialize(TagContext tags, OutputStream output) throws IOException;
+  public abstract byte[] toByteArray(TagContext tags);
```

Comment:
Is this what the gRPC team was asking for? In the past, they had an issue with this approach because it requires always creating a new byte[] rather than appending to a given one e.g. appendToByteArray(tags, byte[], int offset) - that adds its own set of problems (e.g. if the input array isn't long enough) but is a concern for client code where performance is a major concern.

Reply:
I tried to match the BinaryFormat API from the tracing package: https://github.com/census-instrumentation/opencensus-java/blob/bf42c2c695ba80eef4b94745b187b0d553309f0d/api/src/main/java/io/opencensus/trace/propagation/BinaryFormat.java

I'll cc @zhangkun83 .

Reply:
Returning a `byte[]` is not an issue for gRPC, because [Metadata.BinaryMarshaller's serialization method](https://github.com/grpc/grpc-java/blob/v1.6.1/core/src/main/java/io/grpc/Metadata.java#L76) returns a `byte[]` too.

Reply:
if writing to an offset is an issue, we could overload later, keeping simpler for now.